docs/guides/tfhers/pregenerated-key.md

```diff
@@ -0,0 +1,16 @@
+# Second Use Case: Pregenerated TFHE-rs keys
+
+The second scenario involves two parties with a predetermined set of TFHE-rs keysets. The goal is to perform computations on encrypted TFHE-rs values in Concrete. In this case, there is no shared secret key. The party using Concrete will utilize TFHE-rs public keys only and will need to optimize to the parameters of those keys. The party using TFHE-rs will have the capability to perform encryption, decryption, and computation. The flow of this use case is as follows:
+
+1. We generate multiple keysets (secret key, and public keys) with different set of parameters
+2. From these public keys (parameters), Concrete Optimizer will limit its search space to find the best parameters which are compatible with existing TFHE-rs keys
+
+![alt text](../../_static/tfhers/tfhers_uc2.excalidraw.svg)
+
+This scenario is interesting in the case of threshold cryptography, when the secret key is not known by a single party but split into different parties, and when the secret key cannot be changed (e.g. because it's a permanent key of an encrypted blockchain).
+
+This use case is not managed currently by Concrete and will be available later.
+
```

Comment:
In this scenario, two parties have a predetermined set of TFHE-rs keysets. The goal is to compute on encrypted TFHE-rs values in Concrete, but without sharing any secret key. Instead, the party using Concrete will rely only on TFHE-rs public keys and will optimize its operations according to the parameters associated with those keys. Meanwhile, the party using TFHE-rs will have the capacity to perform encryption, decryption, and computation.

## Workflow
The workflow for this scenario is as follows:

1. We generate multiple keysets (including secret keys and public keys) with different sets of parameters.
2. Using the public keys, Concrete's Optimizer will limit its parameter search space to find the best configuration that is compatible with the existing TFHE-rs keys.

## Use case
This approach is particularly useful in scenarios involving threshold cryptography, where the secret key is distributed across multiple parties, ensuring that no single party has full knowledge of the key. This setup is beneficial when the secret key cannot be modified, such as in situations where the key is a permanent fixture, like in encrypted blockchains.

## Future Support
Currently, this use case is not supported by Concrete but will be available later.